src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -1441,7 +1441,11 @@ void notifyListeners(List<FlowNode> nodes, boolean synchronous) {
                 }
             } finally {
                 if (synchronous) {
-                    bc.abort(); // hack to skip save—we are holding a lock
+                    bc.abort(); //// hack to skip save—we are holding a lock
+                } else if (!(getDurabilityHint().isPersistWithEveryStep()) && !(nodes.stream().anyMatch(x->x instanceof FlowEndNode))) {
+                    // We do not want to invoke gratuitous save calls if we aren't set to persist with every step
+                    // Except that we MUST trigger a persistence call at the very end of the execution
+                    bc.abort();
```

Comment:
We aren't worried about whether this causes JENKINS-50199 to happen more often because even if it does, it should only matter when reloading configuration from disk, and that case is now handled by jenkinsci/workflow-job-plugin#104, right?

Reply:
@dwnusbaum Correct -- and if the cause of JENKINS-50199 is still triggered, then that's a bug somewhere else -- if we were relying on this to trigger persistence, that's a clear bug.